src/battle_4.c

```diff
@@ -1671,7 +1680,7 @@ u8 TypeCalc(u16 move, u8 bank_atk, u8 bank_def)
                     ModulateDmgByType2(gTypeEffectiveness[i + 2], move, &flags);
                 //check type2
                 if (gTypeEffectiveness[i + 1] == gBattleMons[bank_def].type2 &&
-                    gBattleMons[bank_def].type1 != gBattleMons[bank_def].type2)
+                    gBattleMons[gBankTarget /* what the christ */].type1 != gBattleMons[bank_def].type2)
```

Comment:
It's supposed to be checking to see if the mon is single-type or not by seeing if its types are the same, but isn't indexing with the same variable. It happens to work fine to my knowledge because bank_def should always be the same as gBankTarget but I would not be surprised if there's some obscure AI bug because of this.